src/NuGet.Jobs.GitHubIndexer/GitRepoSearchers/GitHubSearcher.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Octokit;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace NuGet.Jobs.GitHubIndexer
+{
+    public class GitHubSearcher : IGitRepoSearcher
+    {
+        public const int MIN_STARS = 100;
+        public const int RESULTS_PER_PAGE = 100; // Maximum is 100 :(
+        private IGitHubClient _client;
+
+        public GitHubSearcher(IGitHubClient client)
```

Comment:
This should accept an `ILogger`:

```suggestion
        public GitHubSearcher(IGitHubClient client, ILogger<GitHubSearcher> logger)
```

You can use this for logging instead of `Console.WriteLine`